http/server_test.ts

```diff
@@ -991,6 +991,44 @@ Deno.test(
   },
 );
 
+Deno.test("Server should not leak async ops when closed", () => {
+  const hostname = "127.0.0.1";
+  const port = 4505;
+  const handler = () => new Response();
+  const server = new Server({ port, hostname, handler });
+  server.listenAndServe();
+  server.close();
```

Comment:
No need to make the change in this PR if this fixes the original issue, but I still think it would be helpful to be able to `await` on `server.close()`.

Reply:
In general I think it makes sense, specially if we want to wait until all the pending handlers are resolved. However, I'm not yet comfortable enough with the codebase to make this change and I would prefer to keep it separate.